### PR TITLE
Chore/Create mock solution for reusability of unittest mocks and data

### PR DESCRIPTION
**Pull request by allinox**
## Description

This PR rips the Mock and Mock-data parts out of the test project and establishes a new project solution 'Mocks'. Mocks is then used by Tests for unit testing as previously, but can now also be reused during runtime in Dev whenever UseMockData is set in appsettings.Development.json

This can be benefitial when developing frontend code for backend endpoints that are not yet created, or are for some reason unavailable or broken. It also lessens the need for setup (souch as Database setup and policy files in Azure) before a developer can begin coding in frontend.

## Related Issue(s)
- #445 

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Mocks/Class1.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Altinn.AccessManagement.UI.Mocks
+{
+    public class Class1
```
Comment (jonkjetiloye) [inline backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Mocks/Class1.cs:3]: I assume this is an unintentionally added class

File: backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Mocks/Altinn.AccessManagement.UI.Mocks.csproj [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    <PackageReference Include="Altinn.Common.PEP" Version="1.3.0" />
+    <PackageReference Include="JWTCookieAuthentication" Version="3.0.1" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\Altinn.AccessManagement.UI.Core\Altinn.AccessManagement.UI.Core.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Folder Include="Data\MaskinportenSchema\Delegation\NewFolder\NewFolder\" />
```
Comment (jonkjetiloye) [inline backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Mocks/Altinn.AccessManagement.UI.Mocks.csproj:30]: NewFolder reference I assume can be removed

File: backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Mocks/Altinn.AccessManagement.UI.Mocks.csproj [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Content Include="Data\**">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </Content>
+  </ItemGroup>
+
+  <ItemGroup>
+    <None Remove="Data\MaskinportenSchema\Delegation\appid-400\from_p50067798\to_810418532\ExpectedOutput_Default.json" />
```
Comment (jonkjetiloye) [inline backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Mocks/Altinn.AccessManagement.UI.Mocks.csproj:16]: can be removed

File: backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Mocks/Class1.cs (hunk 1 above)
Comment (allinox) [reply backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Mocks/Class1.cs:3]: Yup, very much unintentional 😅

File: backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Mocks/Altinn.AccessManagement.UI.Mocks.csproj (hunk 2 above)
Comment (allinox) [reply backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Mocks/Altinn.AccessManagement.UI.Mocks.csproj:30]: Oops 😅
Nice catch!

**Review by jonkjetiloye — APPROVED** [review]

**Pull request merged**